go/app/main.go

```diff
@@ -20,22 +24,116 @@ type Response struct {
 	Message string `json:"message"`
 }
 
+type Item struct {
+	ID 	     int    `json:"id"`
+	Name     string `json:"name"`
+	Category string `json:"category"`
+	Image    string `json:"image_name"`
+}
+
+type Items []Item
+
 func root(c echo.Context) error {
 	res := Response{Message: "Hello, world!"}
 	return c.JSON(http.StatusOK, res)
 }
 
 func addItem(c echo.Context) error {
 	// Get form data
+    id := c.FormValue("id")
+    c.Logger().Infof("Receive id: %s", id)
+
 	name := c.FormValue("name")
 	c.Logger().Infof("Receive item: %s", name)
 
-	message := fmt.Sprintf("item received: %s", name)
+	category := c.FormValue("category")
+	c.Logger().Infof("Receive category: %s", category)
+
+	image, err := c.FormFile("image")
+	if err != nil {
+		c.Logger().Error("Failed to receive image file")
+	}
+	c.Logger().Infof("Receive image: %s", image)
+
+	// Create a SHA256 hash
+	hash := sha256.New()
+
+	src, err := image.Open()
+	if err != nil {
+		c.Logger().Error("Failed to open image file")
+	}
+	defer src.Close()
+
+	if _, err := io.Copy(hash, src); err != nil {
+		c.Logger().Error("Failed to hash image file")
+	}
+
+	src.Seek(0, 0)
+	img_name := fmt.Sprintf("%x", hash.Sum(nil)) + ".jpg"
+
+	// Open the JSON file
+	jsonFile, err := os.ReadFile("./items.json")
+	if err != nil {
+		return err
+	}
+
+	// Decode the JSON file into a Go slice
+	var items Items
+	json.Unmarshal(jsonFile, &items)
+
+	// Convert id from string to int
+	idInt, err := strconv.Atoi(id)
+	if err != nil {
+		return err
+	}
+
+	items = append(items, Item{ID: idInt, Name: name, Category: category, Image: img_name})
+
+	// Encode the slice back into JSON
+	file, _ := json.MarshalIndent(items, "", " ")
+
+	_ = os.WriteFile("./items.json", file, 0644)
+
+	message := fmt.Sprintf("item received: %s", img_name)
 	res := Response{Message: message}
 
 	return c.JSON(http.StatusOK, res)
 }
 
+func getItems(c echo.Context) error {
+	// Open the JSON file
+	jsonFile, err := os.ReadFile("./items.json")
+	if err != nil {
+		return err
+	}
+
+	var items Items
+	json.Unmarshal(jsonFile, &items)
+
+	return c.JSON(http.StatusOK, items)
+}
+
+func getOneItem(c echo.Context) error {
```

Comment:
Consistency is the key in naming methods and variables. In that sense, it would be better to name this as `getItem` to make it consistent with `addItem`.
Representing something as plural by using the plural noun and singular by singular noun is a commonly-used practice.